test/dshell/dwarf.d

```diff
@@ -0,0 +1,80 @@
+import dshell;
+
+import std.algorithm;
+import std.file;
+import std.process;
+
+int main()
+{
+    version(DigitalMars) { }
+    else
+    {
+        writeln("Skipping dwarf.d for non-DMD compilers.");
+        return DISABLED;
+    }
+
+    version(Windows)
+    {
+        writeln("Skipping dwarf.d for Windows.");
+        return DISABLED;
+    }
+
+
+    // If the Unix system doesn't have objdump, disable the tests
+    auto sysHasObjdump = executeShell("objdump --help");
+    if (sysHasObjdump.status)
+        return DISABLED;
+
+    // test them all
+    foreach (string path; dirEntries(EXTRA_FILES ~ "/dwarf/", SpanMode.shallow))
+    {
+        if (!isFile(path) || extension(path) != ".d")
+            continue;
+
+        // retrieve the filename without the extension
+        auto filename = baseName(stripExtension(path));
+
+        string exe = OUTPUT_BASE ~ "/" ~ filename;
+        run("$DMD -m$MODEL -of" ~ exe ~ "$EXE -conf= -fPIE -g"
+            ~ " -I$EXTRA_FILES/dwarf/ " ~ path);
+
+        // Write objdump result to a file
+        auto objdump = OUTPUT_BASE ~ "/" ~ filename ~ ".objdump";
+
+        auto objdump_file = File(objdump, "w");
+        run("objdump -W " ~ exe, objdump_file);
+        objdump_file.close();
+
+        // Objdump excepted results
+        string excepted_results_file = EXTRA_FILES ~ "/dwarf/excepted_results/"
+            ~ filename ~ ".txt";
+
+        if (!exists(excepted_results_file))
+            assert(0, "DWARF tests must have a .txt file in the `excepted_results`"
+                ~ " folder which contains the DWARF dump info");
+
+        // Read file result as a string
+        auto result = cast(string)readText(objdump);
+
+        string failmsg;
+
+        foreach (line; File(excepted_results_file).byLine)
+        {
+            if (!canFind(result, line))
+            {
+                failmsg ~= filename ~ ": Couln't find `" ~ line
+                    ~ "` in the DWARF dump info.\n";
+            }
+        }
+
+        if (failmsg)
+        {
+            // Writes the result into stdout for the CI machines.
+            writeln(result);
+            assert(0, failmsg);
```

Comment:
This still stops after the fist failed tests. Maybe set a flag here to return 0/1 below.